module/multiversionmodule/module.go

```diff
@@ -0,0 +1,117 @@
+// Package main is a module designed to help build tests for reconfiguration logic between module versions
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/edaniels/golog"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+	"go.viam.com/utils"
+
+	"go.viam.com/rdk/components/generic"
+	"go.viam.com/rdk/components/motor"
+	"go.viam.com/rdk/module"
+	"go.viam.com/rdk/resource"
+)
+
+var myModel = resource.NewModel("acme", "demo", "multiversionmodule")
+
+// VERSION is set with `-ldflags "-X main.VERSION=..."`.
+var VERSION string
```

Comment:
Does this have to be all caps/is that idiomatic because we're setting with `-ldflags`?

Reply:
I don't think it has to be this way. I think it's conventional (in general programming, perhaps not Go?) to make global "constants" all caps.

I think there's some value in being all caps. Notably, it can signal that the variable should not be written to (`const VERSION` is apparently not an option[1]). Though I think it's a best practice not to have globals that are writable, neither in Go nor systems in other languages!

[1] `ldflags` does not work with constants. E.g:
```
var B string = "Unset"

func main() {
	const A = "Name:"
	fmt.Println(A, B)
}
```
can have B substituted with an ldflag but `const B = "Unset"` cannot.

Reply:
Gotcha; I'm fine with the `VERSION` naming (I've seen screaming-snake-case global variables in go before IIRC) as long as the linter doesn't complain. I suppose I was just curious whether the ldflag substitution _only_ worked with all-caps variables. Good to know it works with all exported variables but not consts.

Reply:
> I suppose I was just curious whether the ldflag substitution only worked with all-caps variables

Ah, just to clarify/confirm, ldflags does not require all-caps variables.